src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -291,20 +301,24 @@ public void run() {
             }
 
             gogoduck.userLog.log(String.format("Processing file '%s'", file.toPath().getFileName()));
-            applySubsetSingleFileNcks(file, module);
+            gogoduck.applySubsetSingleFileNcks(file, module, ncksSubsetParameters);
         }
     }
 
     private void applySubsetMultiThread(Path tmpDir, GoGoDuckModule module, int threadCount) throws GoGoDuckException {
-        userLog.log(String.format("Applying subset '%s'", module.getNcksSubsetParameters().toString()));
         logger.info(String.format("Applying subset on directory '%s'", tmpDir));
 
         File[] directoryListing = tmpDir.toFile().listFiles();
-        logger.info(String.format("Subset for operation is '%s'", module.getNcksSubsetParameters()));
 
         ExecutorService executorService = Executors.newFixedThreadPool(threadCount);
+        List<String> ncksSubsetParameters = null;
         for (final File file : directoryListing) {
-            executorService.submit(new NcksRunnable(file, module, this));
+            if (ncksSubsetParameters == null) {
+                ncksSubsetParameters = module.getNcksSubsetParameters(file.getAbsoluteFile().toString()).getNcksParameters();
```

Comment:
Why not probe files for collection metadata in one place instead of two i.e. setup and here